packages/scripts/src/virtual-modules-loader.ts

```diff
@@ -0,0 +1,39 @@
+import type webpack from 'webpack';
+
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+
+export interface VirtualModulesLoaderOptions {
+    [fileName: string]: string;
+}
+
+const virtualModulesLoader: webpack.LoaderDefinition = function () {
+    const options = this.getOptions() as VirtualModulesLoaderOptions;
+    const match = /^\?id=(.*)$/.exec(this.resourceQuery);
+    if (!match) {
+        throw new Error(`invalid resourceQuery: ${this.resourceQuery}`);
+    }
+    const fileName = match[1]!;
+    if (!options[fileName]) {
+        throw new Error(`missing generator for: ${fileName}`);
+    }
+    return options[fileName]!;
```

Comment:
I think a function would allow greater control but as long as the text is working I don't mind.